gitlab/data_source_gitlab_users.go

```diff
@@ -117,14 +170,76 @@ func flattenGitlabUsers(users []*gitlab.User) []interface{} {
 			"projects_limit":     user.ProjectsLimit,
 			"state":              user.State,
 			"external":           user.External,
+			"extern_uid":         user.ExternUID,
+			"organization":       user.Organization,
+			"two_factor_enabled": user.TwoFactorEnabled,
 		}
 
 		if user.CreatedAt != nil {
-			values["created_at"] = user.CreatedAt.String()
+			values["created_at"] = user.CreatedAt
 		}
 
 		usersList = append(usersList, values)
 	}
 
 	return usersList
 }
+
+func expandGitlabUsersOptions(d []interface{}) (*gitlab.ListUsersOptions, int, error) {
+	if len(d) == 0 {
+		return nil, 0, nil
+	}
+
+	data := d[0].(map[string]interface{})
+	listUsersOptions := &gitlab.ListUsersOptions{}
+	options := ""
+
+	if orderBy := data["order_by"].(string); orderBy != "" {
+		listUsersOptions.OrderBy = &orderBy
+		options += orderBy
+	}
+	if sort := data["sort"].(string); sort != "" {
+		listUsersOptions.Sort = &sort
+		options += sort
+	}
+	if search := data["search"].(string); search != "" {
+		listUsersOptions.Search = &search
+		options += search
+	}
+	if active := data["active"].(bool); active != false {
+		listUsersOptions.Active = &active
+		options += strconv.FormatBool(active)
+	}
+	if blocked := data["blocked"].(bool); blocked != false {
+		listUsersOptions.Blocked = &blocked
+		options += strconv.FormatBool(blocked)
+	}
+	if externalUID := data["extern_uid"].(string); externalUID != "" {
+		listUsersOptions.ExternalUID = &externalUID
+		options += externalUID
+	}
+	if provider := data["provider"].(string); provider != "" {
+		// listUsersOptions.Provider = &provider
+		options += provider
+	}
+	if createdBefore := data["created_before"].(string); createdBefore != "" {
+		date, err := time.Parse("2006-01-02", createdBefore)
+		if err != nil {
+			return nil, 0, fmt.Errorf("created_before must be in yyyy-mm-dd format")
+		}
+		listUsersOptions.CreatedBefore = &date
+		options += createdBefore
+	}
+	if createdAfter := data["created_after"].(string); createdAfter != "" {
+		// date, err := time.Parse("2006-01-02", createdAfter)
```

Comment:
WIP?

Reply:
The field created_after is not yet implemented in the client, so I commented the code. Should it be better to totally remove it?

Reply:
I see two options:

- fork the client, add it, use the fork here using dep aliasing and un-comment the code
- remove it

Reply:
Got it! Thanks!

Reply:
I just made a PR https://github.com/xanzy/go-gitlab/pull/388.

- [ ] Update vendor when client PR will be merged.

Reply:
It's already been merged so I also made the vendor update